src/commands/ext-install.ts

```diff
@@ -51,60 +51,69 @@ export const command = new Command("ext:install [extensionName]")
+      );
+    }
+    let source: ExtensionSource | undefined;
+    let extensionVersion: ExtensionVersion | undefined;
+    const projectId = getProjectId(options);
     // If the user types in a local path (prefixed with ~/, ../, or ./), install from local source.
     // Otherwise, treat the input as an extension reference and proceed with reference-based installation.
-    if (isLocalPath(extensionName)) {
+    if (isLocalPath(extensionRef)) {
       // TODO(b/228444119): Create source should happen at deploy time.
       // Should parse spec locally so we don't need project ID.
-      source = await createSourceFromLocation(needProjectId({ projectId }), extensionName);
-      await displayExtInfo(extensionName, "", source.spec);
+      source = await createSourceFromLocation(needProjectId({ projectId }), extensionRef);
+      await displayExtensionVersionInfo(source.spec);
       void track("Extension Install", "Install by Source", options.interactive ? 1 : 0);
     } else {
       void track("Extension Install", "Install by Extension Ref", options.interactive ? 1 : 0);
-      extensionName = await canonicalizeRefInput(extensionName);
-      extensionVersion = await extensionsApi.getExtensionVersion(extensionName);
-      await infoExtensionVersion({
-        extensionName,
+      const extension = await extensionsApi.getExtension(extensionRef);
+      const extensionVersionRef = await resolveVersion(refs.parse(extensionRef), extension);
+      extensionVersion = await extensionsApi.getExtensionVersion(extensionVersionRef);
+      await displayExtensionVersionInfo(
+        extensionVersion.spec,
         extensionVersion,
-      });
+        extension.latestApprovedVersion,
+        extension.latestVersion
+      );
+      if (extensionVersion.state === "DEPRECATED") {
+        throw new FirebaseError(
+          `Extension version ${clc.bold(
+            extensionVersionRef
+          )} is deprecated and cannot be installed. To install the latest non-deprecated version, omit the version in the extension ref.`
+        );
+      }
+      logger.info();
+      // Check if selected version is older than the latest approved version, or the latest version only if there is no approved version.
+      if (
+        (extension.latestApprovedVersion &&
+          semver.gt(extension.latestApprovedVersion, extensionVersion.spec.version)) ||
+        (!extension.latestApprovedVersion &&
+          extension.latestVersion &&
+          semver.gt(extension.latestVersion, extensionVersion.spec.version))
+      ) {
+        logger.info(
+          `You are about to install extension version ${clc.bold(
+            extensionVersion.spec.version
+          )} which is older than the latest ${
+            extension.latestApprovedVersion ? "accepted version" : "version"
+          } ${clc.bold(extension.latestApprovedVersion!)}.`
```

Comment:
What happens if latestApprovedVersion is null here? Should we display `extensions.latestVersion` instead of `latestApprovedVersion`?

Reply:
Good catch, fixed!